pkg/controllers/vrep/replication_reconciler.go

```diff
@@ -238,33 +236,67 @@ func (r *ReplicationReconciler) checkSandboxExists() error {
 	return nil
 }
 
+func (r *ReplicationReconciler) getServiceIP(ctx context.Context, svcName, dBname string) (ip string, err error) {
```

Comment:
A simpler solution would be to just check if the service name belongs to one of the subclusters. If that's the case you fetch it and get the ip. You don't even need to check the labels.